Add argument for one hot encoding to parsnip (#332)
tidymodels/parsnip: man/rmd/one-hot.Rmd

```{r load, include = FALSE}
library(dplyr)
```

> By default, `model.matrix()` generates binary indicator variables for all but one of the levels for factor predictors when the formula does not remove an intercept. 

Author: This sentence is so confusing 😩 "for... when..."

Reviewer: Instead, I used 

> By default, `model.matrix()` generates binary indicator variables for factor predictors. When the formula does not remove an intercept, an incomplete set of indicators are created; no indicator is made for the first level of the factor.

Author: Oh boy, I still don't think this is clear. I am pretty sure I knew what you are trying to say and I was all: "Wait, what are incomplete indicators???" What about something like:

> By default, `model.matrix()` generates indicator variables for factor predictors. No indicator variable is created for the first (or baseline) level of the factor.

> For example, `species` and `island` both have three levels:

Author: I just realized the 3 is before the encoding, so we should draw attention to the encoding.

```suggestion
For example, `species` and `island` both have three levels but `model.matrix()` creates two indicator variables for each:
```

```{r ref-cell}
library(dplyr)
library(modeldata)
data(penguins)

model.matrix(~ species + island, data = penguins) %>% 
  colnames()
```

> When there is no intercept, the first factor receives indicators for _all_ factor levels but all other factors get all but one (as above):

Author: ```suggestion
When there is no intercept, the first factor is expanded to indicators for _all_ factor levels but all other factors are expanded to all but one (as above):
```

> When there is no intercept, the first factor is expanded to indicators for _all_ factor levels but all other factors are expanded to all but one (as above):

Author: ```suggestion
For a formula with no intercept, the first factor is expanded to indicators for _all_ factor levels but all other factors are expanded to all but one (as above):
```

```{r hybrid}
model.matrix(~ 0 + species + island, data = penguins) %>% 
  colnames()
```

For inference, this hybrid encoding can be problematic. 

> To generate all indicators, this contrast can be used: 

Author: ```suggestion
To generate all indicators, use this contrast: 
```

```{r load, include = FALSE}
library(dplyr)
```

> By default, `model.matrix()` generates binary indicator variables for factor predictors. When the formula does not remove an intercept, an incomplete set of indicators are created; no indicator is made for the first level of the factor.

Author: ```suggestion
By default, `model.matrix()` generates binary indicator variables for factor predictors. No indicator variable is created for the first (baseline) level of the factor.
```
I still think it's pretty opaque; I was trying to parse "incomplete"??? and feeling confused until I realized it meant like "normal". Can we not mention the intercept until we show removing it?
